src/translator/service.cpp

```diff
@@ -56,45 +56,45 @@ std::vector<Response> Service::translateMultiple(std::vector<std::string> &&inpu
   // We queue the individual Requests so they get compiled at batches to be
   // efficiently translated.
   std::vector<std::future<Response>> responseFutures;
-  for (auto &input : inputs) {
-    std::future<Response> inputResponse = queueRequest(std::move(input), responseOptions);
-    responseFutures.push_back(std::move(inputResponse));
+  for (size_t i = 0; i < inputs.size(); i++) {
+    // https://stackoverflow.com/a/33437008/4565794
+    // @jerinphilip is not proud of this.
+    Ptr<std::promise<Response>> responsePromise = New<std::promise<Response>>();
```

Comment:
The `std::vector<Response>` could be grown to the appropriate size then written to asynchronously by the callbacks without a lock.  Then use a semaphore (or condition variable) to wait for completion.  There's no need for `future` nor a need to fill them.

Reply:
However, were you to make `translateMultiple` exist only for WASM, then there would be no need for this, so I suggest doing that first, provided all regression tests pass.